src/script/ptl-tool.py

```diff
@@ -42,51 +104,62 @@
         m = patt.match(line)
         CONTRIBUTORS[m.group(1)] = m.group(2)
 
-def build_branch(branch_name, pull_requests):
-    repo = git.Repo(".")
+def build_branch(args):
+    base = args.base
+    branch = args.branch
+    label = args.label
+
+    G = git.Repo(args.git)
 
-    repo.remotes.upstream.fetch()
+    # First get the latest base branch and PRs from BASE_REMOTE
+    remote = getattr(G.remotes, BASE_REMOTE)
+    remote.fetch()
 
-    # First get the latest base branch from upstream
-    if branch_name == 'HEAD':
+    if base == 'HEAD':
         log.info("Branch base is HEAD; not checking out!")
     else:
-        if branch_name in SPECIAL_BRANCHES:
-            base = BASE % branch_name
-        else:
-            base = BASE % "master"
-        log.info("Branch base on {}".format(base))
-        base = filter(lambda r: r.path == base, repo.refs)[0]
+        log.info("Detaching HEAD onto base: {}".format(base))
+        try:
+            base_path = args.base_path + base
+            base = filter(lambda r: r.path == base_path, G.refs)[0]
+        except IndexError:
+            log.error("Branch base does not exist!")
+            sys.exit(1)
 
         # So we know that we're not on an old test branch, detach HEAD onto ref:
         base.checkout()
```

Comment:
Please review https://github.com/batrick/ceph/pull/3 and provide your comments. This works for me. Otherwise, it leaves "detached from origin/master".